[benjiBase] iP

benjiBase opened the pull request:
# Nameless
> “Your mind is for having ideas, not holding them.” – David Allen [source](https://dansilvestre.com/productivity-quotes/)

Nameless frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* FAST SUPER FAST to use

All you need to do is,
1. download
2. double-click it
3. add your tasks

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
``` Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Add level-0 function

Commit:
Add level-1 function

Commit:
Add Level-2 features

Commit:
Add Level-3 features

Commit:
Add Level-4 features

Commit:
Add A-TextUiTesting updated txt

Commit:
Add Level-5 features

Commit:
Implmented Level-6 features and updated testcases

Commit:
Add Level-7 Save requirement

Commit:
Merge branch 'branch-Level-7'

Commit:
Add Level-8 DateTime requirement and Exception Handling for it

Commit:
Merge branch 'branch-Level-8'

Commit:
Add A-MoreOOP:Use more OOP

Commit:
Add A-Packages

Commit:
Merge remote-tracking branch 'origin/add-gradle-support'

Commit:
Add A-Gradle

Commit:
Add A-JUnit Test

Commit:
Add A-Jar Create JAR file

Commit:
Add header comments to non-private classes/methods

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Adhere to all java coding standard by course

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Add Level-9 Find implementation

Commit:
Merge branch 'branch-Level-9'

Commit:
Implement more coding convention

ITLimJiaWei commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,31 @@
+
+public class Event extends Task {
+    protected LocalDateTime from;
+    protected LocalDateTime to;
```
from/to could be named startTime/endTime or something that identifies the variables have to do with DateTime

ITLimJiaWei commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,87 @@
+    /**
+     * Show the message printed when a task is deleted.
+     */
+    public void showDeleteTask (TaskList tasks, int index) throws DukeException {
```
Small inconsistency where showDeleteTask and (TaskList ...) have a space in between where most other methods don't. A few other occurrences of this in other methods in this class

ITLimJiaWei commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,118 @@
+     * Creates directory if it does not exist.
+     * @throws DukeException if error creating directory
+     */
+    private void createDirectory() throws  DukeException {
```
Very small typo with a double space before DukeException 😄

ITLimJiaWei commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,118 @@
+            return new Todo(words[2]);
+        } else if(words[0].equals("D")){
+            return new Deadline(words[2], LocalDateTime.parse(words[3], DISPLAY_FORMAT));
+        } else if(words[0].equals("E")){
```
Good job keeping the if else cases neat but there should be a space after the if/else if before the condition
E.g. `if (words[0])`

ITLimJiaWei commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,42 @@
+ * A class that handle storing and dealing of task in ArrayList.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
```
Good job adhering to a plural name for the ArrayList object 👍

ITLimJiaWei commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,87 @@
+ * A class that deals with interactions in terminal.
+ */
+public class Ui {
+    private final String LINE = "______________________________________________________________";
```
Good job adhering to uppercase constant naming 👍

ITLimJiaWei commented on src/test/java/duke/StorageTest.java:
```diff
@@ -0,0 +1,57 @@
+
+    @Test
+    public void testWriteFile() throws DukeException {
+        Storage storage = new Storage("testing/test.txt");
```
You could consider having "testing/test.txt" as a constant string declared before the test cases run to save some lines of code but overall good job 👍

ITLimJiaWei reviewed (commented):
LGTM. Some nits to consider but overall very good job on code quality/standards 😄

Commit:
Add Level-10 GUI

Commit:
Merge branch 'branch-Level-10'

Jollybomber commented on src/main/java/duke/Nameless.java:
Perhaps some of these functions can be parked under the Parser class instead?

Jollybomber commented on src/main/java/duke/Nameless.java:
```diff
@@ -0,0 +1,153 @@
+     * Adds a new todo task when input is "todo"
+     * Prints the output of the added task
+     */
+    private String todo(String input) throws DukeException {
```
I think the naming for these functions could provide more detail.

Jollybomber commented on src/main/java/duke/Nameless.java:
```diff
@@ -0,0 +1,153 @@
+            try {
+                if (input.equals("bye")) {
+                    return bye();
+                } else if (input.equals("list")) {
+                    return ui.showList(tasks);
+                } else if (input.matches("mark \\d+")) {
+                    return ui.showMarkTask(tasks, Parser.splitGetNum(input));
+                } else if (input.matches("unmark \\d+")) {
+                    return ui.showUnmarkTask(tasks, Parser.splitGetNum(input));
+                } else if (input.matches("todo(?: .+)?")) {
+                    return todo(input);
+                } else if (input.matches("deadline(?: .+)?")) {
+                    return deadline(input);
+                } else if (input.matches("event(?: .+)?")) {
+                    return event(input);
+                } else if (input.matches("delete \\d+")) {
+                    return ui.showDeleteTask(tasks, Parser.splitGetNum(input));
+                } else if (input.matches("find(?: .+)?")) {
+                    return find(input);
```
Nice job with the parsing of the input using regex!

Jollybomber commented on src/main/java/duke/Exception/DukeException.java:
```diff
@@ -0,0 +1,14 @@
+package duke.Exception;
+
+public class DukeException extends Exception {
```
Perhaps there could have been more exception classes to handle different types of errors? Your TypeOfException is a good example for the Duke specific formatting erorrs.

Jollybomber reviewed (commented):
Good job, LGTM!

Commit:
Add assertions for Storage and Nameless class

Used assertions in the Nameless class to
ensure that the inputs are not null in the
todo, deadline and event methods. The assertions
will help to ensure that users cannot input null.

Used assertions in the Storage class to verify
the creation of directory and file in the
createDirectory and createFile methods. The
assert statements will ensure that file and
directory exists after program execution.

Commit:
Merge branch 'branch-A-Assertions'

Commit:
Improve Code Quality of codebase

Critically examines the code and
refactors it to improve the overall code quality.

Key improvements include:
* Sturcture code logically and use blank lines
* Ensure expressions are clear and simplified
* Ensure magic numbers are replaced with constants

benjiBase committed:
Merge pull request #1 from benjiBase/branch-A-CodeQuality

Improve Code Quality of codebase

Commit:
Implement B-ViewSchedule feature

Users can enter view <date> in format of
yyyy-mm-dd to view the schedule of that day.
This feature enhances the task management
capabilities of the application by allowing users
to view their tasks in a schedule format.

Commit:
Add UI picture in docs

Commit:
Update mark/unmark output

benjiBase committed:
Update README.md
